File: backend/app/schemas/billing.py
from pydantic import BaseModel


class BillingPreviewRow(BaseModel):
    source_type: str
    sequence: str
    customer_number: str
    business_number: str
    company_name: str
    manager_name: str
    base_date: str
    first_login: str
    latest_login: str
    login_count: int
    billing_company_name: str
    bank_company_name: str
    match_status: str
    note: str


class BillingPreviewSummary(BaseModel):
    total_count: int
    matched_count: int
    name_mismatch_count: int
    missing_count: int
    open_count: int
    erp_count: int


class BillingPreview(BaseModel):
    spreadsheet_title: str
    spreadsheet_url: str
    generated_from: list[str]
    rows: list[BillingPreviewRow]
    summary: BillingPreviewSummary

File: frontend/src/api/billing.ts
import { apiClient } from './client';

export type BillingPreviewRow = {
  source_type: string;
  sequence: string;
  customer_number: string;
  business_number: string;
  company_name: string;
  manager_name: string;
  base_date: string;
  first_login: string;
  latest_login: string;
  login_count: number;
  billing_company_name: string;
  bank_company_name: string;
  match_status: string;
  note: string;
};

export type BillingPreviewSummary = {
  total_count: number;
  matched_count: number;
  name_mismatch_count: number;
  missing_count: number;
  open_count: number;
  erp_count: number;
};

export type BillingPreview = {
  spreadsheet_title: string;
  spreadsheet_url: string;
  generated_from: string[];
  rows: BillingPreviewRow[];
  summary: BillingPreviewSummary;
};

export async function fetchBillingPreview() {
  const response = await apiClient.get<BillingPreview>('/billing/preview');
  return response.data;
}

export async function uploadBillingLoginFile(file: File) {
  const formData = new FormData();
  formData.append('login_file', file);
  const response = await apiClient.post<BillingPreview>('/billing/preview', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  });
  return response.data;
}
